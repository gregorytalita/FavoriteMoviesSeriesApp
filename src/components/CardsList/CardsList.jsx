import React from 'react'
import { Card } from '../'
import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles(() => ({
  list: {
    width: '80%',
    display: 'flex',
    justifyContent: 'space-around',
    flexWrap: 'wrap'
  }
}))

const CardsList = ({ favorites, removeFromFavorites }) => {
  const classes = useStyles()

  return (
    <div className={classes.list}>
      {
        favorites.map((favorite, index) => (
          <Card key={index} {...favorite} remove={() => removeFromFavorites(favorite)} />
        ))
      }
    </div>
  )
}

export default CardsList
