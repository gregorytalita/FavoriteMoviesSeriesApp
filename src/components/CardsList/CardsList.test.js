import React from 'react'
import { create } from 'react-test-renderer'
import CardsList from './CardsList'

const ListData = [{
  media_type: 'movie',
  poster_path: '/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg',
  release_date: '2019-03-06',
  title: 'Captain Marvel',
  user: 'talitagregory@hotmail.com',
  vote_average: 7
}]

describe('CardsList component', () => {
  test('it matches the snapshot', () => {
    const component = create(<CardsList favorites={ListData} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
