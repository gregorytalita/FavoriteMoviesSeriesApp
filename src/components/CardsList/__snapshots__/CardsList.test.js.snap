// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList component it matches the snapshot 1`] = `
<div
  className="makeStyles-list-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation3 card MuiPaper-rounded"
    style={
      Object {
        "background": "url(https://image.tmdb.org/t/p/w300//AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg)",
      }
    }
  >
    <div
      className="cardOverlay"
    />
    <div
      className="cardRate"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        <svg>
          rate.svg
        </svg>
         
        7
      </p>
      <svg>
        movie.svg
      </svg>
    </div>
    <div
      className="cardContent"
    >
      <div
        className="cardInfo"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
        >
          <b>
            Captain Marvel
          </b>
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          2019-03-06
        </p>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-button-2 MuiButton-text MuiButton-textSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg>
            remove.svg
          </svg>
           Remove
        </span>
      </button>
    </div>
  </div>
</div>
`;
